astropy/nddata/nddata_base.py

```diff
@@ -14,79 +14,66 @@
 @six.add_metaclass(ABCMeta)
 class NDDataBase(object):
     """
-    Base metaclass that defines the interface for NDData
+    Base metaclass that defines the interface for N-dimensional datasets with
+    associated meta informations used in ``astropy``.
 
-    Classes that wish to use this interface without inheriting from
-    `~astropy.nddata.NDData` should subclass ``NDDataBase`` instead.
-
-    All properties and methods except uncertainty must be override by derived
-    classes.
+    NDDataBase makes no attempt of restricting any attribute or defines how
+    these are stored. Therefore all properties and methods have to be
+    overridden in subclasses. See `~astropy.nddata.NDData` for a subclass that
```

Comment:
Suggested edit: drop the sentence that begins "NDDataBase makes...". 

Change "Therefore all" -> "All"